### DISCUSSION TITLE
Prefect Server fails at creating Postgres Database

### DISCUSSION THREAD

**Discussion by den-pro-soft** [PrefectHQ/prefect#3570] [General]
Hello, Thanks for this wonderful tool.

I am trying to run the prefect server locally.

I have installed `prefect` by using `conda` and run the server with `prefect server start`.

Then, it gives me the below error.

```
WARNING: The PREFECT_SERVER_DB_CMD variable is not set. Defaulting to a blank string.
WARNING: The DB_CONNECTION_URL variable is not set. Defaulting to a blank string.
WARNING: The POSTGRES_DB variable is not set. Defaulting to a blank string.
WARNING: The POSTGRES_PASSWORD variable is not set. Defaulting to a blank string.
WARNING: The POSTGRES_USER variable is not set. Defaulting to a blank string.
Creating network "prefect-server" with the default driver
Creating tmp_postgres_1 ... done
Creating tmp_hasura_1   ... done
Creating tmp_graphql_1  ... done
Creating tmp_towel_1    ... done
Creating tmp_apollo_1   ... done
Creating tmp_ui_1       ... done
Attaching to tmp_postgres_1, tmp_hasura_1, tmp_graphql_1, tmp_towel_1, tmp_apollo_1, tmp_ui_1
graphql_1   | bash: -c: line 0: syntax error near unexpected token `&&'
graphql_1   | bash: -c: line 0: ` && python src/prefect_server/services/graphql/server.py'
hasura_1    | {"type":"pg-client","timestamp":"2020-10-26T11:45:30.504+0000","level":"warn","detail":{"message":"postgres connection failed, retrying(0)."}}
hasura_1    | {"type":"pg-client","timestamp":"2020-10-26T11:45:30.504+0000","level":"warn","detail":{"message":"postgres connection failed, retrying(1)."}}
hasura_1    | {"type":"startup","timestamp":"2020-10-26T11:45:30.504+0000","level":"error","detail":{"kind":"db_migrate","info":{"internal":"could not connect to server: No such file or directory\n\tIs the server running locally and accepting\n\tconnections on Unix domain socket \"/var/run/postgresql/.s.PGSQL.5432\"?\n","path":"$","error":"connection error","code":"postgres-error"}}}
ui_1        | 👾👾👾 UI running at localhost:8080 👾👾👾
postgres_1  | Error: Database is uninitialized and superuser password is not specified.
postgres_1  |        You must specify POSTGRES_PASSWORD to a non-empty value for the
postgres_1  |        superuser. For example, "-e POSTGRES_PASSWORD=password" on "docker run".
postgres_1  | 
postgres_1  |        You may also use "POSTGRES_HOST_AUTH_METHOD=trust" to allow all
postgres_1  |        connections without a password. This is *not* recommended.
postgres_1  | 
postgres_1  |        See PostgreSQL documentation about "trust":
postgres_1  |        https://www.postgresql.org/docs/current/auth-trust.html
tmp_hasura_1 exited with code 1
tmp_graphql_1 exited with code 1
tmp_postgres_1 exited with code 1
```
I have done nothing. Just installed the `prefect` and started.

I am not sure what's going on and will be appreciated your help.
Thanks

**Comment by joshmeek** [comment]
Hi @aj-super, those warnings at the top look like a culprit for the error as it seems like it isn't picking up the default configuration variables.

Couple questions:
- What OS are you on?
- Are you running `prefect server start` in the same conda environment where it is installed? (i.e. not in another terminal session without the activated environment)

**Comment by den-pro-soft** [comment]
Hi @joshmeek. Thanks for your response.
- My OS is the Linux OS( Ubuntu 18.04 )
- Of course, I ran the prefect server in the same conda environment.

**Reply by den-pro-soft** [reply]
When I run the `postgres` docker image directly using the bellow command, it works.
    **```docker run --rm  --name pg-docker -e POSTGRES_PASSWORD=password -p 5432:5432 postgres```**
But, running `prefect server start` doesn't work because of the issue in creating postgres container.
what do you say?

**Reply by joshmeek** [reply]
This looks similar to #3557 ([comment on patch](https://github.com/PrefectHQ/prefect/issues/3557#issuecomment-716070507)). It seems like Ubuntu is not respecting the environment that is passed to the subprocess so a `.env` file is needed in that case to set these environment variables. Will get this prioritized and fixed 👍

**Reply by den-pro-soft** [reply]
Okay. Thanks. I wish it would be fixed as soon as possible. So that I can run the prefect UI on my local machine.
